ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -165,18 +169,53 @@ public void sendRequest(
       requestBuilder.tag(requestId);
     }
 
-    OkHttpClient client = mClient;
+    final RCTDeviceEventEmitter eventEmitter = getEventEmitter(executorToken);
+    OkHttpClient.Builder clientBuilder = mClient.newBuilder();
+
+    // If JS is listening for progress updates, install a ProgressResponseBody that intercepts the
+    // response and counts bytes received.
+    if (useIncrementalUpdates) {
+      clientBuilder.addNetworkInterceptor(new Interceptor() {
+        @Override
+        public Response intercept(Interceptor.Chain chain) throws IOException {
+          Response originalResponse = chain.proceed(chain.request());
+          ProgressResponseBody responseBody = new ProgressResponseBody(
+            originalResponse.body(), new ProgressListener() {
```

Comment:
Formatting around here seems wrong. Should either be:

```
... = new ProgressResponseBody(..., ... {
  long last = ...
```
if it doesn't exceed 100 chars width, or
```
... = new ProgressResponseBody(
  ...,
  ... {
    long last = ...
```